app/api/tests/test_api.py

```diff
@@ -831,6 +831,51 @@ def doCleanups(cls):
         remove_all_role_mappings()
 
 
+def TestTagAPI(APITestcase):
+
+    @classmethod
+    def setUpTestData(cls):
+        cls.main_project_member_name = 'project_member_name'
+        cls.main_project_member_pass = 'project_member_pass'
+        cls.sub_project_member_name = 'sub_project_member_name'
+        cls.sub_project_member_pass = 'sub_project_member_pass'
+        cls.approver_name = 'approver_name_name'
+        cls.approver_pass = 'approver_pass'
+        cls.super_user_name = 'super_user_name'
+        cls.super_user_pass = 'super_user_pass'
+        create_default_roles()
+        main_project_member = User.objects.create_user(username=cls.main_project_member_name,
+                                                       password=cls.main_project_member_pass)
+        sub_project_member = User.objects.create_user(username=cls.sub_project_member_name,
+                                                      password=cls.sub_project_member_pass)
+        approver = User.objects.create_user(username=cls.approver_name,
+                                            password=cls.approver_pass)
+        User.objects.create_superuser(username=cls.super_user_name,
+                                      password=cls.super_user_pass,
+                                      email='fizz@buzz.com')
+        cls.main_project = mommy.make('TextClassificationProject', users=[main_project_member])
+        cls.sub_project = mommy.make('TextClassificationProject', users=[sub_project_member])
+        assign_user_to_role(project_member=main_project_member, project=cls.main_project,
+                            role_name=settings.ROLE_ANNOTATOR)
+        assign_user_to_role(project_member=sub_project_member, project=cls.sub_project,
+                            role_name=settings.ROLE_ANNOTATOR)
+        assign_user_to_role(project_member=approver, project=cls.main_project,
+                            role_name=settings.ROLE_ANNOTATION_APPROVER)
+        cls.url = reverse(viewname='tag_list')
+        cls.tag = mommy.make('Tag', project=cls.main_project, text='Tag 1')
+
+
+    def test_create_tag(self):
+        self.client.login(username=self.main_project_member_name,
+                        password=self.main_project_member_pass)
+        response = self.client.post(self.url, data={'text': 'Tag 2'})
+        self.assertEqual(response.status_code, status.HTTP_201_CREATED)
```

Comment:
add a test to see if tags are listed for a project and another one for tag deletion